scan/vulnerability.go

```diff
@@ -0,0 +1,119 @@
+package scan
+
+import (
+	"crypto/rsa"
+	"crypto/x509"
+	"errors"
+	"net/url"
+
+	"github.com/FiloSottile/Heartbleed/heartbleed"
+	"github.com/cloudflare/cfssl/certinfo"
+	"github.com/cloudflare/cfssl/helpers"
+)
+
+// Vulnerability contains scanners to detect vulnerabilities such as Heartbleed
+var Vulnerability = &Family{
+	Description: "Scans a host to detect TLS vulnerabilities.",
+	Scanners: map[string]*Scanner{
+		"Heartbleed": {
+			"Scan for Heartbleed vulnerability",
+			heartbleedScan,
+		},
+		"BERserk": {
+			"Scan for BERserk vulnerability",
+			BERserkScan,
+		},
+		"LogJam": {
+			"Scan for LogJam vulnerability",
+			LogJamScan,
+		},
+	},
+}
+
+func heartbleedScan(addr, hostname string) (grade Grade, output Output, err error) {
+	tgt := &heartbleed.Target{
+		Service: "https",
+		HostIp:  hostname,
+	}
+
+	u, err := url.Parse(tgt.HostIp)
+	if err == nil && u.Host != "" {
+		tgt.HostIp = u.Host
+	}
+
+	out, err := heartbleed.Heartbleed(tgt,
+		[]byte("github.com/FiloSottile/Heartbleed"), true)
+	if err == heartbleed.Safe {
+		grade = Good
+		err = nil
+	} else if err != nil {
+		grade = Warning
+	} else {
+		grade, output = Bad, out
+	}
+	return
+}
+
+func BERserkScan(addr, hostname string) (grade Grade, output Output, err error) {
```

Comment:
looks like Travis failed here, since this function name's first letter is capitalized, add a comment above it or uncapitalized it. Reference: http://www.goinggo.net/2014/03/exportedunexported-identifiers-in-go.html